### PR TITLE
Adding to CI the Security Baseline module test and disabling from build and test unnecessary modules

### PR DESCRIPTION
**Pull request by MariusNi**
## Description

Adding to CI the Security Baseline module test and disabling from build and test unnecessary modules. Keeping only: CommandRunner, DeviceInfo, Configuration and SecurityBaseline. All other modules were found to either don't work on Mariner or not have any scenario.

Now that the Security Baseline test recipe test is expanded to exercise in full all remediations and audits, we are also enabling this test (alongside the ones for remaining enabled modules) to PR CI and to E2E.

Also, starting stabilization for remediation with this occasion (work in progress, to be continued in other PRs)

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by AhmedBM — APPROVED** [review]
👌

**Review by robertschaedler3 — APPROVED** [review]

**Review by robertschaedler3 — APPROVED** [review]

File: src/modules/test/main.c [hunk 1]
```diff
@@ -501,11 +527,33 @@ int RunTestStep(const TEST_STEP* test, const MANAGEMENT_MODULE* module)
     else if (test->type == DESIRED)
     {
         mmiStatus = module->set(module->session, test->component, test->object, test->payload, test->payloadSize);
-
+
```
Comment (AhmedBM) [inline src/modules/test/main.c:530]: nit: spacing

**Review by AhmedBM — APPROVED** [review]
Looks good, just minor nit 👌

File: src/modules/test/main.c (hunk 1 above)
Comment (MariusNi) [reply src/modules/test/main.c:530]: Thanks for the feedback

**Pull request merged**